### DISCUSSION TITLE
How can I provide custom compile options to OpenImageIO?

### DISCUSSION THREAD

**Discussion by MarkCallow** [microsoft/vcpkg#26725] [Q&A]
I want to use OpenImageIO but building it and its dependencies takes 30 minutes or more. I only need support for a handful of formats: exr, jpeg, jpeg2000, png and webp. OIIO has pre-processor defines that can be used to omit formats.

Can I provide my own vcpkg.json that somehow sets these CPP defines and removes no-longer needed dependencies?

I'm working on a portable app so I really want to use a package manager rather than building myself for each platform.

**Comment by Neumann-A** [comment]
You can use overlay ports https://github.com/microsoft/vcpkg/blob/master/docs/specifications/ports-overlay.md to customize your OpenImageIO build as you want. (considering that whatever you want is not exported as features of the port and OIIO has some features.)

Although there might be CPP options to control the build I would still advice to use cmake options which OIIO. You might be able to leverage `VCPKG_CMAKE_CONFIGURE_OPTIONS` from your triplet without using an overlay.

**Reply by MarkCallow** [reply]
Wow! thanks for the fast response. Unfortunately OIIO does not have cmake options equivalent to the compile options for omitting plug-ins. I will study the overlay port documentation.
